packages/mask/src/social-network-adaptor/instagram.com/collecting/identity.ts

```diff
@@ -0,0 +1,65 @@
+import { LiveSelector, MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
+import { ProfileIdentifier, delay } from '@masknet/shared-base'
+import { creator, SocialNetworkUI as Next } from '../../../social-network'
+import Services from '../../../extension/service'
+import { instagramBase } from '../base'
+import { searchAvatarSelector } from '../utils/selector'
+import { getAvatar, getBioDescription, getNickname, getPersonalHomepage, getUserId } from '../utils/user'
+
+function resolveCurrentVisitingIdentityInner(
+    ref: Next.CollectingCapabilities.IdentityResolveProvider['recognized'],
+    cancel: AbortSignal,
+) {
+    const avatarSelector = searchAvatarSelector()
+    const assign = async () => {
+        await delay(500)
+        const bio = getBioDescription()
+        const homepage = getPersonalHomepage()
+        const nickname = getNickname()
+        const handle = getUserId()
+        const avatar = getAvatar()
+
+        ref.value = {
+            identifier: new ProfileIdentifier(instagramBase.networkIdentifier, handle),
+            nickname,
+            avatar,
+            bio,
+        }
+        Services.Helper.resolveTCOLink(homepage).then((link) => {
```

Comment:
wait, Instagram is also using `t.co` short link? I thought it was only for Twitter